force-app/main/adapter/in/sobjects/contact/ContactService.cls

```diff
@@ -307,4 +324,32 @@ public with sharing class ContactService {
             index++;
         }
     }
+
+    private Boolean isOneToOne(Account accountRecord) {
+        return accountRecord.npe01__SYSTEM_AccountType__c
+                == CAO_Constants.ONE_TO_ONE_ORGANIZATION_TYPE;
+    }
+
+    private Boolean isHousehold(Account accountRecord) {
+        return accountRecord.npe01__SYSTEM_AccountType__c
+                == CAO_Constants.HH_ACCOUNT_TYPE;
+    }
+
+    private List<Id> getContactIds(List<Contact> contactsWithoutAccounts) {
+        List<Id> contactIds = new List<Id>();
+        for (Contact contactRecord : contactsWithoutAccounts) {
+            if (contactRecord.Id != null) {
+                contactIds.add(contactRecord.Id);
+            }
+        }
+        return contactIds;
+    }
+
+    private List<Account> getAccounts(List<Id> contactIds) {
```

Comment:
I think we should move this into an `HouseHoldSelector` class. Or better still we let the `ContactAdapter` collect this data since it's already collaborating with the selector already.